src/platform/Linux/bluez/Helper.cpp

```diff
@@ -387,33 +386,29 @@ static gboolean BluezCharacteristicWriteFD(GIOChannel * aChannel, GIOCondition a
 
     ChipLogDetail(DeviceLayer, "c1 %s mtu, %d", __func__, conn->mMtu);
 
-    buf = static_cast<gchar *>(g_malloc(conn->mMtu));
-    fd  = g_io_channel_unix_get_fd(aChannel);
-
-    len = read(fd, buf, conn->mMtu);
-
+    buf = new uint8_t[conn->mMtu];
```

Comment:
nit: typically `Platform::MemoryAlloc` is used in Matter stack to easily replace the allocator. Also, you could check if the allocation succeeded.

Reply:
As for now changed to `g_new` to be consistent with the rest of the platform code.

In the future we might try to revise the Linux platform code and see whether it will be possible not to use `g_malloc()` family (it asserts for malloc success, so there is no need to check for NULL). However, I doubt that it will be possible, since glib internally uses `g_malloc()` everywhere. So almost all (if not specified otherwise) glib functions will abort the application in case of malloc failure...

Reply:
Makes sense!